src/sql/src/plan/statement/ddl.rs

```diff
@@ -3177,7 +3185,12 @@ pub fn plan_create_cluster_replica(
         sql_bail!("cannot create more than one replica of a cluster containing sources or sinks");
     }
     ensure_cluster_is_not_linked(scx, cluster.id())?;
-    ensure_cluster_is_not_managed(scx, cluster.id())?;
```

Comment:
Why can this go away?

Reply:
We're validating this when sequencing the drop operation. The issue is that here we can't know if a replica is internal or not, so we have to delay this check -- or we expose additional information to the SQL layer through the `CatalogClusterReplica` trait.